### DISCUSSION TITLE
How to embed/shortcode for youtube/vimeo/soundcloud/spotify and other audio/video contents?

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [welpo/tabi#289] [Q&A] [answered]
Hi @welpo 

first things first: thank you for tabi! it is impressive!

secondly:

- does tabi come with shortcodes for audio/video contents? (single video and playlists alike)
  - if yes: how do I use them?
  - if not: could you provide and example to extend tabi?

Please and thank you

ps: btw this is where I am using tabi: https://gacallea.github.io

**Comment by welpo** [comment, marked as answer]
Hello Andrea!

Thank you for the kind words.

tabi does not include media embed shortcodes.

The simplest way to embed content is adding the HTML embed code in your MarkDown files.

You will need to [configure (or disable)](https://welpo.github.io/tabi/blog/mastering-tabi-settings/#csp-content-security-policy) the CSP to allow connections to SoundCloud, YouTube, and any site you want to embed content from.

Hope that helps!

**Reply by andreacfromtheapp** [reply]
thanks!!! :)
